parsers/regex/regex.go

```diff
@@ -0,0 +1,158 @@
+// Package regex consumes logs given user-defined regex format for lines
+
+/* Example usage (from honeytail root)
+  		go install . && \
+ 			honeytail -p regex -k $HONEYTAIL_WRITEKEY \
+			-f /private/var/system.log \
+			--dataset 'MY_TEST_DATASET' \
+			--backfill \
+			--regex.line_regex="(?P<time>\d{2}:\d{2}:\d{2})" \
+			--regex.timefield="time" \
+			--regex.time_format="%H:%M:%S"
+*/
+
+package regex
+
+import (
+	"errors"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/honeycombio/honeytail/event"
+	"github.com/honeycombio/honeytail/httime"
+	"github.com/honeycombio/honeytail/parsers"
+)
+
+type Options struct {
+	LineRegex       string `long:"line_regex" description:"a regular expression with named capture groups representing the fields you want parsed"`
+	TimeFieldName   string `long:"timefield" description:"Name of the field that contains a timestamp"`
+	TimeFieldFormat string `long:"time_format" description:"Timestamp format to use (strftime and Golang time.Parse supported)"`
+
+	NumParsers int `hidden:"true" description:"number of regex parsers to spin up"`
+}
+
+type Parser struct {
+	conf       Options
+	lineParser LineParser
+}
+
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
+
+	if p.conf.LineRegex != "" {
```

Comment:
Do we also want to consider it an error if `LineRegex` is empty here? Seems like it wouldn't be very useful to run honeytail with `--parser=regex` if you don't also specify a regex.